### PR TITLE
Reverse `PtrAdd` inlining

### PR DESCRIPTION
**Pull request by ltratt**
This commit is the first nudge towards "reverse code generation by proxy". After a code-shuffling commit, the main meat is in 1b02a254.

### CODE REVIEW

**Comment by ltratt** [comment]
Needs https://github.com/ykjit/yk/pull/1484 to be merged first.

File: ykrt/src/compile/jitc_yk/jit_ir/mod.rs [hunk 1]
```diff
@@ -2158,30 +2160,32 @@ impl LoadInst {
 ///
 /// Specifies the [yksmp::Location] of a run-time argument.
 #[derive(Clone, Copy, Debug)]
-#[repr(packed)]
 pub struct ParamInst {
-    /// The [yksmp::Location] of this input.
-    locidx: u32,
+    /// The [InstIdx] of the [yksmp::Location] parameter.
+    paramidx: InstIdx,
```
Comment (vext01) [inline ykrt/src/compile/jitc_yk/jit_ir/mod.rs:2147]: I've gotten confused by this. Are you are reusing `InstIdx` for indices of a vector other than `self.insts`? If that's what's happening, shouldn't we introduce a `ParamIdx`?

Comment (vext01) [reply ykrt/src/compile/jitc_yk/jit_ir/mod.rs:2147]: I see. It's a little confusing still. Up to you guys.

Comment (ltratt) [reply ykrt/src/compile/jitc_yk/jit_ir/mod.rs:2147]: Basically (and I think this is sensible!) parameters come first, contiguously. `ParamIdx` slightly obfuscates this. Could we do better? Maybe. But I think that's for a future PR.

**Comment by ltratt** [comment]
Rebased against 1484, so this should merge OK once 1484 has merged.

**Comment by ltratt** [comment]
Had to rebase against master: force pushed a rebase.

**Pull request merged**